bip-blind-merged-mining.mediawiki

```diff
@@ -0,0 +1,341 @@
+
+This data is "critical" in the sense that it is the minimum amount of data required to define a sidechain.
+
+Where does this data come from, and how does it get around?
+
+==== Creating / Broadcasting This Data ====
+
+===== Creation =====
+
+By the time Blind Merged Mining can take place, the ChainIndex is globally known (it is the "Account Number" in D1 [see previous BIP], and "nSidechain" in the code). Each sidechain, when activated by soft fork, will take one of the 0-255 available indexes.
+
+The other two items, sideHeaderHash and prevBlockRef, are created by sidechain nodes. sideHeaderHash is quite straightforward -- side:nodes build side:blocks, and take the hash of these.
+
+The final item, prevBlockRef, is a little more complicated. It is an integer that counts the number of "skips" one must take in the side:chain in order to find the current side:block's parent block. In practice, this value will usually be zero. It will only be a value other than zero, in cases where invalid sidechain blocks have been mined, or when a side:node intentionally wants to orphan some side:blocks (if a side:node wants to orphan the most-recent N blocks, the value of the current block will be equal to N ; in the block after that it will be back to zero).
+
+<img src="bip-blind-merged-mining/bmm-dots-examples.png?raw=true" align="middle"></img>
+
+Since the hashes themselves are already ordered by the mainchain, tracing the blockchain's path by index (prevBlockRef) will be the same as tracing it by identifying a list of hashes. In other words, the ordering given via each side:block's "prevBlockRef" will be isomorphic to an ordering given by each side:block's "prevSideHeaderHash" ... if "prevSideHeaderHash is defined to be the sidechain's equivalent of the mainchain's "prevBlockHash". It will be possible to freely convert from one to the other. See M8 to learn more about how these hashes are requested by sidechain block creators to be included in the mainchain.
+
+Now that we know what our critical data is, and how it is made, how is this data broadcast and stored?
+
+===== Broadcast =====
+
+Mainchain nodes are going to need this data later, so it must be easy to find. We will put it into the coinbase via OP RETURN.
+
+==== M7 -- "Blind-Mine the Sidechain(s)" ====
+
+Thus, (for n sidechains) we have a coinbase output with multiple OP_RETURNs (we've changed the tx-standardness policy to allow multiple OP_RETURNs):
```

Comment:
There is no tx-standardness policy for coinbases (it is a self-contradicting concept)... Nor are such matters of policy subject to BIPs - they are entirely up to implementations and/or individual node operators.

Reply:
That may be so, but in practice many people have pointed out to me that I am "not allowed" to use multiple OP Returns. So it definitely interests some readers.

And we did in fact change the standardness policy.

Should we ensure that we have only changed the policy for coinbase txns only?

Reply:
Policy is about off-chain transaction acceptance. There is no policy for on-chain blocks - just consensus rules.

Reply:
Ah, I see what you are saying.

I will try to find a way to clarify the sentence.